src/main/java/dev/jbang/dependencies/DependencyAttributes.java

```diff
@@ -0,0 +1,61 @@
+package dev.jbang.dependencies;
+
+import dev.jbang.util.AttributeParser;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class DependencyAttributes {
+
+	private final Map<String, List<String>> attributes;
+
+	// Java 8 Set.of() is sad :)
+	static Set<String> defaultScopes = Collections.unmodifiableSet(new HashSet<>(Arrays.asList("build", "run")));
+	Set<String> scopes;
+
+	public DependencyAttributes(Map<String, List<String>> attributes) {
+		this.attributes = attributes;
+	}
+
+	// lazy computed; dont store hashset unless truly needed
+	public boolean includeInScope(String scope) {
+		if (scopes == null) {
+			if (attributes.containsKey("scope")) {
+				scopes = new HashSet<>(attributes.get("scope"));
+			} else {
+				scopes = defaultScopes;
+			}
+		}
+		return scopes.contains(scope);
+	}
+
+	/**
+	 * is there something configured? even if default ?
+	 * @return
+	 */
+	public boolean isDefault() {
+		return !attributes.isEmpty();
+	}
+
+	public String toStringFormat() {
+		return AttributeParser.toStringRep(attributes, "scope");
+	}
+
+	@Override
+	public String toString() {
+		return toStringFormat();
+	}
+
+	@Override
+	public boolean equals(Object o) {
```

Comment:
so first push resulted in funky TestModules errors that was caused by equals method missing on Attributes - which is fair..but reveals a conondrum - how do we define equals for these? i've implemented a few tests (which currently fails) showing situations that we probably want to make equals...which we can by unifing attributes...but should we?

Reply:
just did an update where i split it out in a DependencyRequest even though I do thing the other properties makes sense to have on the "thing" that encodes the dependency - but it seems like we have places where we treat MavenCoordinate as absolutes which can be used as keys....that is NOT the case even with these new info because MavenCoordinates can have  version ranges defined.